web-portal/internal-api/upload-managers/LocalUploader.js

```diff
@@ -0,0 +1,21 @@
+const fs = require('fs')
+const { join, resolve } = require('path')
+
+const LOCAL_UPLOAD_ROOT = resolve(__dirname, '../../static')
+
+module.exports = class LocalUploader {
+  constructor(urlBase) {
+    this.urlBase = urlBase
+  }
+
+  upload(path, data) {
+    return new Promise((resolve, reject) => {
+      // wx flag mitigates the possibility of clobbering an existing file
+      // (will error out on write)
+      fs.writeFile(join(LOCAL_UPLOAD_ROOT, path), data, { flag: 'wx' }, (err) => {
```

Comment:
This is fine but I wonder fs.Promises is available in the node version we use? If so we could possibly simplify this

Reply:
Looks like it's added in v14. Might refactor to use it after we're certain we've upgraded production infrastructure.